### PR TITLE
remap sms and status urls, log people still calling old ones

### PR DESCRIPTION
**Pull request by nicpottier**
Tested on my box, this will remap all the URLs we can and start logging those still calling us so we can track those down manually.

### CODE REVIEW

File: temba/channels/types/twilio/views.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def claim_number(self, user, phone_number, country, role):
         # create new TwiML app
         callback_domain = org.get_brand_domain()
         new_receive_url = "https://" + callback_domain + reverse('courier.t', args=[channel_uuid, 'receive'])
-        new_status_url = "https://" + callback_domain + reverse('courier.t', args=[channel_uuid, 'status'])
+        new_status_url = "https://" + callback_domain + reverse('handlers.twilio_handler', args=['status', channel_uuid])
```
Comment (nicpottier) [inline temba/channels/types/twilio/views.py]: The status URL is actually for calls, not SMS, so shouldn't be going to Courier.

File: temba/channels/handlers.py [hunk 2]
```diff
@@ -81,9 +81,6 @@ def all_subclasses(cls):
 
 class TwimlAPIHandler(BaseChannelHandler):
 
-    courier_url = r'^tw/(?P<uuid>[a-z0-9\-]+)/(?P<action>receive|status)$'
-    courier_name = 'courier.tw'
```
Comment (nicpottier) [inline temba/channels/handlers.py]: This is now handled through the `courier_url` attribute on the channel type.

File: temba/channels/migrations/0087_migrate_twilio_endpoints.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        return
+
+    for channel in Channel.objects.filter(is_active=True, channel_type__in=['T', 'TMS']).select_related('org'):
+        if 'account_sid' not in channel.config or 'auth_token' not in channel.config or 'application_sid' not in channel.config:
+            print("Skipping app migration for %s, missing settings in %s" % (channel.uuid, channel.config))
+
+        try:
+            prefix = "https://" + channel.callback_domain
+
+            sms_url = prefix + reverse('courier.t', args=[channel.uuid, 'receive'])
```
Comment (norkans7) [inline temba/channels/migrations/0087_migrate_twilio_endpoints.py]: TMS receive on courier.tms I think

Comment (nicpottier) [reply temba/channels/migrations/0087_migrate_twilio_endpoints.py]: Oooh, great catch!

Comment (nicpottier) [reply temba/channels/migrations/0087_migrate_twilio_endpoints.py]: Actually, as far as I can tell this doesn't actually apply for TMS apps.. they are handled manually by the users.

File: temba/channels/types/twilio/views.py (hunk 1 above)
Comment (norkans7) [reply temba/channels/types/twilio/views.py]: I think this status URL is for msgs, IVR uses ivr.ivrcall_handle

Comment (nicpottier) [reply temba/channels/types/twilio/views.py]: No, its not. I checked into that quite a bit. Callbacks for messages are specified at sending time. This attribute in the API is actually the voice status callback.

File: temba/channels/migrations/0087_migrate_twilio_endpoints.py (hunk 3 above)
Comment (norkans7) [reply temba/channels/migrations/0087_migrate_twilio_endpoints.py]: right

File: temba/channels/types/twilio/views.py (hunk 1 above)
Comment (norkans7) [reply temba/channels/types/twilio/views.py]: https://github.com/nyaruka/courier/blob/master/handlers/twilio/twilio.go#L181

Comment (nicpottier) [reply temba/channels/types/twilio/views.py]: That's a different URL. The one above is configured for voice on the application. Go check you Twilio account and the app settings, you'll see.

Comment (norkans7) [reply temba/channels/types/twilio/views.py]: Yep I get it now

File: temba/channels/migrations/0087_migrate_twilio_endpoints.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+from django.db import migrations
+from django.urls import reverse
+from twilio.rest import TwilioRestClient
+
+
+def update_application_urls(Channel):
+    if not settings.IS_PROD:
+        return
+
+    for channel in Channel.objects.filter(is_active=True, channel_type='T').select_related('org'):
```
Comment (norkans7) [inline temba/channels/migrations/0087_migrate_twilio_endpoints.py]: why the select_related ?

Comment (nicpottier) [reply temba/channels/migrations/0087_migrate_twilio_endpoints.py]: callback URL might call into the org.. (its not a real attribute, its a function)

**Pull request merged**